src/librustc_resolve/lib.rs

```diff
@@ -99,16 +99,276 @@ use std::usize;
+    /// error E0410: variable from pattern is not bound in pattern #1
+    VariableNotBoundInParentPattern(syntax::ast::Name, usize),
+    /// error E0411: use of `Self` outside of an impl or trait
+    SelfUsedOutsideImplOrTrait,
+    /// error E0412: use of undeclared
+    UseOfUndeclared(&'b str, &'b str),
+    /// error E0413: declaration shadows an enum variant or unit-like struct in scope
+    DeclarationShadowsEnumVariantOrUnitLikeStruct(syntax::ast::Name),
+    /// error E0414: only irrefutable patterns allowed here
+    OnlyIrrefutablePatternsAllowedHere,
+    /// error E0415: identifier is bound more than once in this parameter list
+    IdentifierBoundMoreThanOnceInParameterList(&'b str),
+    /// error E0416: identifier is bound more than once in the same pattern
+    IdentifierBoundMoreThanOnceInSamePattern(&'b str),
+    /// error E0417: static variables cannot be referenced in a pattern
+    StaticVariableReference,
+    /// error E0418: is not an enum variant, struct or const
+    NotAnEnumVariantStructOrConst(&'b str),
+    /// error E0419: unresolved enum variant, struct or const
+    UnresolvedEnumVariantStructOrConst(&'b str),
+    /// error E0420: is not an associated const
+    NotAnAssociatedConst(&'b str),
+    /// error E0421: unresolved associated const
+    UnresolvedAssociatedConst(&'b str),
+    /// error E0422: does not name a struct
+    DoesNotNameAStruct(&'b str),
+    /// error E0423: is a struct variant name, but this expression uses it like a function name
+    StructVariantUsedAsFunction(&'b str),
+    /// error E0424: `self` is not available in a static method
+    SelfNotAvailableInStaticMethod,
+    /// error E0425: unresolved name
+    UnresolvedName(&'b str, &'b str),
+    /// error E0426: use of undeclared label
+    UndeclaredLabel(&'b str),
+    /// error E0427: cannot use `ref` binding mode with ...
+    CannotUseRefBindingModeWith(&'b str),
+    /// error E0428: duplicate definition
+    DuplicateDefinition(&'b str, &'b str),
+    /// error E0429: `self` imports are only allowed within a { } list
+    SelfImportsOnlyAllowedWithin,
+    /// error E0430: `self` import can only appear once in the list
+    SelfImportCanOnlyAppearOnceInTheList,
+    /// error E0431: `self` import can only appear in an import list with a non-empty prefix
+    SelfImportOnlyInImportListWithNonEmptyPrefix,
+    /// error E0432: unresolved import
+    UnresolvedImport(Option<(&'b str, Option<&'b str>)>),
+    /// error E0433: failed to resolve
+    FailedToResolve(&'b str),
+    /// error E0434: can't capture dynamic environment in a fn item
+    CannotCaptureDynamicEnvironmentInFnItem,
+    /// error E0435: attempt to use a non-constant value in a constant
+    AttemptToUseNonConstantValueInConstant,
+}
+
+fn resolve_error<'b, 'a:'b, 'tcx:'a>(resolver: &'b Resolver<'a, 'tcx>, span: syntax::codemap::Span,
+                                     resolution_error: &ResolutionError<'b>) {
```

Comment:
What @pnkfelix was suggesting, I believe, was to have `if !resolver.emit_errors { return; }` at the start of this function, and use `span_err!` directly.